pkg/apis/workflow/v1alpha1/anystring.go

```diff
@@ -12,40 +12,45 @@ import (
 // * It's JSON type is just string, not `int-or-string`.
 // * It will unmarshall int64 (rather than only int32) and represents it as string.
 // * It will marshall back to string - marshalling is not symmetric.
-type Int64OrString string
+type AnyString string
 
-func ParseInt64OrString(val interface{}) Int64OrString {
-	return Int64OrString(fmt.Sprintf("%v", val))
+func ParseAnyString(val interface{}) AnyString {
+	return AnyString(fmt.Sprintf("%v", val))
 }
 
-func Int64OrStringPtr(val interface{}) *Int64OrString {
-	i := ParseInt64OrString(val)
+func AnyStringPtr(val interface{}) *AnyString {
+	i := ParseAnyString(val)
 	return &i
 }
 
-func (i *Int64OrString) UnmarshalJSON(value []byte) error {
+func (i *AnyString) UnmarshalJSON(value []byte) error {
 	if value[0] == '"' {
 		v := ""
 		err := json.Unmarshal(value, &v)
 		if err != nil {
 			return err
 		}
-		*i = Int64OrString(v)
+		*i = AnyString(v)
 		return nil
 	}
-	v := 0
+	var v interface{}
 	err := json.Unmarshal(value, &v)
 	if err != nil {
 		return err
 	}
-	*i = Int64OrString(strconv.Itoa(v))
+	switch v := v.(type) {
```

Comment:
could this switch handle the string case too?

Reply:
Seems so.